src/main/java/com/Elessar/app/server/KafkaMsgSender.java

```diff
@@ -40,14 +47,21 @@ public P2PMsgResponse send(List<Message> messages, String topic) throws Exceptio
                                 .setTimestamp(message.getTimestamp()));
         }
 
+        topic = topic.replaceAll("\\s", "");
         final ProducerRecord<Long, String> record = new ProducerRecord<>(topic, p2pMsgRequest.toString());
-        final RecordMetadata metadata = producer.send(record).get();
+        final P2PMsgResponse.Builder p2pMsgResponse = P2PMsgResponse.newBuilder();
+        final RecordMetadata metadata = producer.send(record).get();    // Synchronous send and wait for ack before proceed
 
-        logger.trace("Message sent to Kafka server at topic " + topic + " partition " + metadata.partition() + " with offset " + metadata.offset());
+        logger.trace("Message sent to Kafka server at topic " + metadata.topic() + " partition " + metadata.partition() + " with offset " + metadata.offset());
+        p2pMsgResponse.setSuccess(true).setIsDelivered(true);
```

Comment:
Please add a comment here explaining why we set the delivered flag to true, for example what the assumptions we make and which cases we don't handle and so on as this is an important design decision.